src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2660,13 +2735,24 @@ CheckConnectionTimeout(WorkerPool *workerPool)
 			 */
 			WorkerPoolFailed(workerPool);
 
-			/*
-			 * The enforcement is not always erroring out. For example, if a SELECT task
-			 * has two different placements, we'd warn the user, fail the pool and continue
-			 * with the next placement.
-			 */
-			if (execution->transactionProperties->errorOnAnyFailure || execution->failed)
+			if (workerPool->failureState == WORKER_POOL_FAILED_OVER_TO_LOCAL)
+			{
+				/*
+				 *
+				 * When the pool is failed over to local execution, warning
+				 * the user just creates chatter as the executor is capable of
+				 * finishing the execution.
+				 */
+				logLevel = DEBUG1;
+			}
```

Comment:
It looks like we don't have coverage for failure tests, because we trigger this line. I'll try to open a PR similar to this: https://github.com/citusdata/citus/pull/4377/files